### PR TITLE
chore(ai): add theming for AIConversation

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adds design token support and component theme support for AIConversation

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/tokens/components/aiConversation.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      fontSize: { value: '{fontSizes.small.value}' },
+      fontWeight: { value: '{fontWeights.normal.value}' },
+    },
+
+    remove: {
+      padding: { value: '{space.xxs.value}' },
+    },
+
+    image: {
+      width: { value: '{fontSizes.medium.value}' },
```
Comment (reesscot) [inline packages/ui/src/theme/tokens/components/aiConversation.ts:116]: Why would the image size match the font sizes?

File: examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import outputs from './amplify_outputs';
+import type { Schema } from '@environments/ai/gen2/amplify/data/resource';
+import {
+  View,
+  createTheme,
+  defaultDarkModeOverride,
+  withAuthenticator,
+} from '@aws-amplify/ui-react';
+import { ColorMode } from '@aws-amplify/ui-react';
+import { Button } from '@aws-amplify/ui-react';
```
Comment (reesscot) [inline examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx:16]: could combine these imports

File: examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { Amplify } from 'aws-amplify';
+import { createAIHooks, AIConversation } from '@aws-amplify/ui-react-ai';
```
Comment (reesscot) [inline examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx:3]: nit: cleanup these imports so they are grouped by third party, first party, local imports

File: .changeset/sharp-buttons-look.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui": patch
+---
+
+chore(ai): adds theming and design tokens for AIConversation component
```
Comment (reesscot) [inline .changeset/sharp-buttons-look.md:5]: Isn't this redundant with the more detailed changeset above?

File: examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        handleSendMessage={sendMessage}
+        isLoading={isLoading}
+        variant="bubble"
+      />
+    </>
+  );
+}
+
+function AIConversationThemePage() {
+  const [colorMode, setColorMode] = React.useState<ColorMode>('light');
```
Comment (reesscot) [inline examples/next/pages/ui/components/ai/ai-conversation-theming-v2/index.page.tsx:98]: At some point maybe we should just offer a hook for this

**Review by reesscot — COMMENTED** [review]
Left some questions/comments

File: packages/ui/src/theme/tokens/components/aiConversation.ts (hunk 1 above)
Comment (dbanksdesign) [reply packages/ui/src/theme/tokens/components/aiConversation.ts:116]: The image here is the image inside the attachment so we want it to be the size of the font size:

CleanShot 2024-10-10 at 13 49 27@2x

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**